src/main/java/org/folio/rest/impl/ItemStorageDereferencedAPI.java

```diff
@@ -0,0 +1,198 @@
+
+  @Validate
+  @Override
+  public void getItemStorageDereferencedItems(
+    int offset, int limit, String query, String lang,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    List<DereferencedItem> mappedResults = new ArrayList<>();
+    String whereClause = "";
+
+    if (query != null) {
+      try {
+        CQLWrapper wrapper = new CQLWrapper(
+          new CQL2PgJSON(ITEM_TABLE + "." + JSON_COLUMN), query, limit, offset);
+        logger.info("translated SQL query: " + wrapper.toString());
+        whereClause = wrapper.toString();
+      } catch(Exception e) {
+        respondWith400Error("Invalid CQL query: " + e.getMessage(), asyncResultHandler);
+      }
+    } else {
+      whereClause = "LIMIT " + limit + " OFFSET " + offset;
+    }
+
+    PostgresClient postgresClient = PgUtil.postgresClient(vertxContext, okapiHeaders);
+    postgresClient.select(sqlQuery + whereClause, asyncResult -> {
+
+      handleSelectFailure(asyncResult, asyncResultHandler);
+
+      asyncResult.result().forEach(row -> {mappedResults.add(mapToDereferencedItem(row));});
+
+      DereferencedItems itemCollection = new DereferencedItems();
+      itemCollection.setDereferencedItems(mappedResults);
+      itemCollection.setTotalRecords(mappedResults.size());
+
+      asyncResultHandler.handle(Future.succeededFuture(
+        GetItemStorageDereferencedItemsResponse.respond200WithApplicationJson(itemCollection)));
+    });
+  }
+
+  @Validate
+  @Override
+  public void getItemStorageDereferencedItemsByItemId(
+    String itemId, String lang, java.util.Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+    logger.info("id number: " + itemId);
+    if (!UuidUtil.isUuid(itemId)) {
+      respondWith400Error("Invalid UUID", asyncResultHandler);
+    }
+    String whereClause = "WHERE item.id='" + itemId + "'";
+    PostgresClient postgresClient = PgUtil.postgresClient(vertxContext, okapiHeaders);
+    postgresClient.select(sqlQuery + whereClause, asyncResult -> {
+      handleSelectFailure(asyncResult, asyncResultHandler);
+      Row row = asyncResult.result().iterator().next();
+      DereferencedItem item = mapToDereferencedItem(row);
+      asyncResultHandler.handle(Future.succeededFuture(
+        GetItemStorageDereferencedItemsByItemIdResponse.respond200WithApplicationJson(item)));
+    });
+  }
+
+  private void handleSelectFailure(
+    AsyncResult<RowSet<Row>> asyncResult,
+    Handler<AsyncResult<Response>> asyncResultHandler) {
+    
+    if (asyncResult.failed()) {
+      String errorMessage = asyncResult.cause().getMessage();
+      respondWith500Error("Can't retrive item records: " + errorMessage, asyncResultHandler);
+    }
+    if (asyncResult.result().size() == 0) {
+      respondWith404Error("No item records found.", asyncResultHandler);
+    }
```

Comment:
Why is getting no record back a failure?

Reply:
I think we should be consistent with the other APIs and return 404 for the single record endpoint and 200 with zero records for the collection endpoint.

Reply:
Apologies, I was under the impression that the standard response for "I didn't find anything" was a 404.  I'll fix.